x-pack/plugins/apm/public/components/shared/TransactionDurationAnomalyAlertTrigger/index.tsx

```diff
@@ -0,0 +1,153 @@
+import { useServiceTransactionTypes } from '../../../hooks/useServiceTransactionTypes';
+import { useUrlParams } from '../../../hooks/useUrlParams';
+import { ServiceAlertTrigger } from '../ServiceAlertTrigger';
+import { PopoverExpression } from '../ServiceAlertTrigger/PopoverExpression';
+import { SelectAnomalySeverity } from './SelectAnomalySeverity';
+
+interface Params {
+  windowSize: number;
+  windowUnit: string;
+  serviceName: string;
+  transactionType: string;
+  environment: string;
+  anomalyScore: 0 | 25 | 50 | 75;
+}
+
+interface Props {
+  alertParams: Params;
+  setAlertParams: (key: string, value: any) => void;
+  setAlertProperty: (key: string, value: any) => void;
+}
+
+export function TransactionDurationAnomalyAlertTrigger(props: Props) {
+  const { setAlertParams, alertParams, setAlertProperty } = props;
+  const { urlParams } = useUrlParams();
+  const transactionTypes = useServiceTransactionTypes(urlParams);
+  const { serviceName, start, end } = urlParams;
+  const { environmentOptions } = useEnvironments({ serviceName, start, end });
+
+  if (!transactionTypes.length || !serviceName) {
+    return null;
+  }
+
+  const defaults: Params = {
+    windowSize: 15,
+    windowUnit: 'm',
+    transactionType: transactionTypes[0],
+    serviceName,
+    environment: urlParams.environment || ALL_OPTION.value,
+    anomalyScore: 75,
+  };
+
+  const params = {
+    ...defaults,
+    ...alertParams,
+  };
+
+  const fields = [
+    <EuiExpression
+      description={i18n.translate(
+        'xpack.apm.transactionDurationAnomalyAlertTrigger.service',
+        {
+          defaultMessage: 'Service',
+        }
+      )}
+      value={
+        <EuiText className="eui-displayInlineBlock">
+          <h5>{serviceName}</h5>
+        </EuiText>
+      }
+    />,
+    <PopoverExpression
+      value={
+        params.environment === ALL_OPTION.value
+          ? ALL_OPTION.text
+          : params.environment
+      }
+      title={i18n.translate(
+        'xpack.apm.transactionDurationAnomalyAlertTrigger.environment',
+        {
+          defaultMessage: 'Environment',
+        }
+      )}
+    >
+      <EuiSelect
+        value={params.environment}
+        options={environmentOptions}
+        onChange={(e) =>
+          setAlertParams('environment', e.target.value as Params['environment'])
+        }
+        compressed
+      />
+    </PopoverExpression>,
+    <PopoverExpression
+      value={params.transactionType}
+      title={i18n.translate(
+        'xpack.apm.transactionDurationAnomalyAlertTrigger.type',
+        {
+          defaultMessage: 'Type',
+        }
+      )}
+    >
+      <EuiSelect
+        value={params.transactionType}
+        options={transactionTypes.map((key) => {
+          return {
+            text: key,
+            value: key,
+          };
+        })}
+        onChange={(e) =>
+          setAlertParams(
+            'transactionType',
+            e.target.value as Params['transactionType']
```

Comment:
Would be nice to avoid `as Params['transactionType']`. Is it because `e.target.value` can be other types than `string`?

Reply:
We can probably safely avoid the type assertion. It's a holdover from [`TransactionDurationAlertTrigger`](https://github.com/elastic/kibana/blob/ec08dcae074a283b6be7877fa177bf8ab2b66db9/x-pack/plugins/apm/public/components/shared/TransactionDurationAlertTrigger/index.tsx#L105) which this component was based on.